src/EditorFeatures/Core.Wpf/InlineRename/UI/Dashboard/RenameDashboard.xaml.cs

```diff
@@ -320,7 +321,13 @@ private void OnTextViewLostAggregateFocus(object sender, EventArgs e)
 
         private void CloseButton_Click(object sender, RoutedEventArgs e)
         {
-            _model.Session.Cancel();
+            _ = CancelAsync();
+        }
+
+        private async Task CancelAsync()
+        {
+            //.ConfigureAwait(true) because we need to set focus later
+            await _model.Session.CancelAsync().ConfigureAwait(true);
             _textView.VisualElement.Focus();
```

Comment:
why not set focus on the _textView *prior* to cancelling the session (which seems like it could take unbounded time).

Reply:
stepping back (and maybe this pr will calrify it), it's unclear to me why *cancelling* (which just should cancel some CT tokens, and return the buffers to the initial state) would be async.  IT seems like it should be fast and UI oriented.  not sure why cancelling itself is async.

Reply:
Ok.  I've made a different change here: https://github.com/dotnet/roslyn/pull/74948

Cancellation here *should* be synchronous here.